richtextfx/src/main/java/org/fxmisc/richtext/model/Paragraph.java

```diff
@@ -132,7 +132,8 @@ public String substring(int from) {
             int segIdx = pos.getMajor();
             List<SEG> segs = new ArrayList<>(segIdx + 1);
             segs.addAll(segments.subList(0, segIdx));
-            segs.add(segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor()));
+            segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor())
+                            .ifPresent(seg -> segs.add(seg));
```

Comment:
@JordanMartinez I have added a test case which, obviously, fails now - https://travis-ci.org/TomasMikula/RichTextFX/builds/171061050

The relevant area is in `Paragraph.subSequence(int start)` (the `trim()` method also needs to be fixed, but that should be trivial once we have the solution for `subSequence()`):
```
...
            segmentOps.subSequence(segments.get(segIdx), pos.getMinor())
                            .ifPresent(segs::add);
            segs.addAll(segments.subList(segIdx + 1, segments.size()));
            if (segs.isEmpty()) {
                segs.add(????);
            }
            return new Paragraph<>(paragraphStyle, segmentOps, segs);
...
```
I am unclear what to add to the `segs` list, in case the result list is empty - we would require an object of type `S` with a length of `0`, essentially ...
Alternatively we might decide that a Paragraph can now also contain an empty list of segments, but since the segment list is exposed through the API, that might influence existing applications ...